### ISSUE TITLE
Support Casting to Derived Entity Types

### ISSUE THREAD

**Issue by texttechne** [odata2ts/odata2ts#330]
Example:
```
### cast to subtype and filter thereby
GET {{baseUrl}}/People('russellwhyte')/Trips(0)/PlanItems/Microsoft.OData.SampleService.Models.TripPin.Flight
```

See [discussion](https://github.com/odata2ts/odata2ts/discussions/320) and [related issue](#327)

**Comment by texttechne** [comment]
### Provided Solution
```ts
mainService.people("russellwhyte").trip(0).planItems().asFlightService().query();
```

**Comment by texttechne** [comment]
Released with [@odata2ts/odata2ts@0.39.0](https://github.com/odata2ts/odata2ts/releases/tag/%40odata2ts%2Fodata2ts%400.39.0)
